src/java.desktop/share/classes/javax/swing/text/html/AccessibleHTML.java

```diff
@@ -183,6 +183,9 @@ private void buildInfo() {
             Element root = doc.getDefaultRootElement();
 
             rootElementInfo = new ElementInfo(root);
+            if (rootHTMLAccessibleContext != null) {
+                rootHTMLAccessibleContext.setElementInfo(rootElementInfo);
+            }
             rootElementInfo.validate();
```

Comment:
probably we should call validate first, so it will be in the valid state before being leaked to the "setElementInfo" method.

Reply:
I think this can be verified by the test, you can check the number of children returned by the a11y API is correct when the doc is replaced.

Reply:
> I think this can be verified by the test, you can check the number of children returned by the a11y API is correct when the doc is replaced.

Probably we don't required a test because I think we can verify the number of children by using JAccessWalker a11y tool, so whenever there is a doc change, the component hierarchy should be updated.

Reply:
> probably we should call validate first, so it will be in the valid state before being leaked to the "setElementInfo" method.

Ok, I will check.

Reply:
> probably we should call validate first, so it will be in the valid state before being leaked to the "setElementInfo" method.

Updated.

Reply:
> I think this can be verified by the test, you can check the number of children returned by the a11y API is correct when the doc is replaced.

Added an automated test to verify the fix. Along with test, few html files and support files are also pushed. 
Please review the latest changes.

Reply:
Please fix all that "Whitespace error" in the test.

Reply:
> Please fix all that "Whitespace error" in the test.
Fixed.